mirrord/config/src/external_proxy.rs

```diff
@@ -0,0 +1,110 @@
+use std::{
+    net::SocketAddr,
+    path::{Path, PathBuf},
+};
+
+use mirrord_config_derive::MirrordConfig;
+use schemars::JsonSchema;
+use serde::Serialize;
+
+use crate::config::source::MirrordConfigSource;
+
+/// Configuration for the external proxy mirrord spawns for mirrord container support, this proxy is
+/// used to allow intproxy running in sidecar to connect to agent
+///
+/// This is seldom used, but if you get `ConnectionRefused` errors, you might
+/// want to increase the timeouts a bit.
+///
+/// ```json
+/// {
+///   "external_proxy": {
+///     "start_idle_timeout": 30,
+///     "idle_timeout": 5
+///   }
+/// }
+/// ```
+#[derive(MirrordConfig, Clone, Debug, Serialize)]
+#[config(map_to = "ExternalProxyFileConfig", derive = "JsonSchema")]
+#[cfg_attr(test, config(derive = "PartialEq"))]
+pub struct ExternalProxyConfig {
+    /// ```json
+    /// {
+    ///   "exyernal_proxy": {
+    ///     "connect_tcp": "10.10.0.100:7777"
+    ///   }
+    /// }
+    /// ```
+    #[config(env = "MIRRORD_EXTERNAL_CONNECT_TCP")]
+    pub connect_tcp: Option<SocketAddr>,
+
+    /// ### external_proxy.client_tls_certificate {#external_proxy-client_tls_certificate}
+    #[config(env = "MIRRORD_EXTERNAL_CLIENT_TLS_CERTIFICATE")]
+    pub client_tls_certificate: Option<PathBuf>,
+
+    /// ### external_proxy.client_tls_key {#external_proxy-client_tls_key}
+    #[config(env = "MIRRORD_EXTERNAL_CLIENT_TLS_KEY")]
+    pub client_tls_key: Option<PathBuf>,
+
+    /// ### external_proxy.tls_certificate {#external_proxy-tls_certificate}
+    #[config(env = "MIRRORD_EXTERNAL_TLS_CERTIFICATE")]
+    pub tls_certificate: Option<PathBuf>,
+
+    /// ### external_proxy.tls_key {#external_proxy-tls_key}
+    #[config(env = "MIRRORD_EXTERNAL_TLS_KEY")]
+    pub tls_key: Option<PathBuf>,
+
+    /// ### external_proxy.start_idle_timeout {#external_proxy-start_idle_timeout}
+    ///
+    /// How much time to wait for the first connection to the proxy in seconds.
+    ///
+    /// Common cases would be running with dlv or any other debugger, which sets a breakpoint
+    /// on process execution, delaying the layer startup and connection to proxy.
+    ///
+    /// ```json
+    /// {
+    ///   "external_proxy": {
+    ///     "start_idle_timeout": 60
+    ///   }
+    /// }
+    /// ```
+    #[config(default = 60)]
+    pub start_idle_timeout: u64,
+
+    /// ### external_proxy.log_level {#external_proxy-log_level}
+    /// Set the log level for the external proxy.
+    /// RUST_LOG convention (i.e `mirrord=trace`)
+    /// will only be used if log_destination is set
+    pub log_level: Option<String>,
+
+    /// ### external_proxy.log_destination {#external_proxy-log_destination}
+    /// Set the log file destination for the external proxy.
+    pub log_destination: Option<String>,
+}
+
+fn env_iter<'a>(
```

Comment:
This looks a bit weird, so I would add docs along the lines of:

Converts an environment variable key `env` and its value (a `Path`) into an `Iterator` of tuples in the form of (key, value) to be used by (tls setup thingy?).